ingest/src/main/scala/hydra.ingest/http/SchemasEndpoint.scala

```diff
@@ -26,34 +26,70 @@ import akka.util.Timeout
 import ch.megard.akka.http.cors.scaladsl.CorsDirectives._
 import hydra.avro.resource.SchemaResource
 import hydra.common.config.ConfigSupport
+import hydra.common.config.ConfigSupport._
 import hydra.core.akka.SchemaRegistryActor
 import hydra.core.akka.SchemaRegistryActor._
 import hydra.core.http.{CorsSupport, RouteSupport}
 import hydra.core.marshallers.GenericServiceResponse
 import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException
 import org.apache.avro.SchemaParseException
-import spray.json.RootJsonFormat
+import spray.json.{JsArray, JsObject, JsValue, RootJsonFormat}
 import hydra.core.monitor.HydraMetrics.addPromHttpMetric
+import hydra.kafka.consumer.KafkaConsumerProxy.{ListTopics, ListTopicsResponse}
+import org.apache.kafka.common.PartitionInfo
+import scalacache.cachingF
+import scalacache._
+import scalacache.guava.GuavaCache
+import scalacache.modes.scalaFuture._
+import akka.http.scaladsl.server.ExceptionHandler
 
+import scala.concurrent.{ExecutionContext, Future}
+import scala.collection.immutable.Map
+import scala.concurrent.Future
 import scala.concurrent.duration._
 
 /**
   * A wrapper around Confluent's schema registry that facilitates schema registration and retrieval.
   *
   * Created by alexsilva on 2/13/16.
   */
-class SchemasEndpoint()(implicit system: ActorSystem)
+class SchemasEndpoint(consumerProxy: ActorSelection)(implicit system: ActorSystem)
     extends RouteSupport
     with ConfigSupport
     with CorsSupport {
 
+  private implicit val cache = GuavaCache[Map[String, Seq[PartitionInfo]]]
+
+
   implicit val endpointFormat: RootJsonFormat[SchemasEndpointResponse] = jsonFormat3(SchemasEndpointResponse.apply)
   implicit val v2EndpointFormat: RootJsonFormat[SchemasWithKeyEndpointResponse] = jsonFormat2(SchemasWithKeyEndpointResponse.apply)
+  implicit val schemasWithTopicFormat: RootJsonFormat[SchemasWithTopicResponse] = jsonFormat2(SchemasWithTopicResponse.apply)
+  implicit object batchSchemasFormat extends RootJsonFormat[BatchSchemasResponse] {
```

Comment:
For some reason I couldn't get this version to work... not sure if there's a better way to do this. 

`implicit val bsf: RootJsonFormat[BatchSchemasResponse] = jsonFormat1(BatchSchemasResponse.apply)`

Reply:
The reason this probably didn't work is that you need to have a similar Json format for all classes that are used within this class. In this case, `BatchSchemasResponse` uses `SchemasWithTopicResponse` which uses `SchemasEndpointResponse`. So you would need a json format available for each of those.